app/src/main/java/co/nimblehq/alpine/sample/MainActivity.kt

```diff
@@ -1,12 +1,80 @@
 package co.nimblehq.alpine.sample
 
-import androidx.appcompat.app.AppCompatActivity
+import android.Manifest.permission.CAMERA
+import android.content.Intent
+import android.content.pm.PackageManager.PERMISSION_GRANTED
+import android.net.Uri
 import android.os.Bundle
+import android.provider.Settings
+import android.view.View
+import android.widget.Button
+import android.widget.Toast
+import androidx.activity.ComponentActivity
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AlertDialog
+import androidx.core.content.ContextCompat.checkSelfPermission
 import co.nimblehq.alpine.R
 
-class MainActivity : AppCompatActivity() {
+class MainActivity : ComponentActivity(), View.OnClickListener {
+
+    private val requestPermissionLauncher =
+        registerForActivityResult(
+            ActivityResultContracts.RequestPermission()
+        ) { isGranted: Boolean ->
+            if (isGranted) {
+                navigateToCamera()
+            } else {
+                showRationale()
+            }
+        }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        listOf<Button>(
+            findViewById(R.id.btn_main_start),
+            findViewById(R.id.btn_main_enter_manually)
+        ).forEach { it.setOnClickListener(this) }
+    }
+
+    override fun onClick(view: View) {
+        when (view.id) {
+            R.id.btn_main_start -> checkCameraPermissionAndNavigateToCamera()
+            R.id.btn_main_enter_manually -> {
+                // TODO: navigate to the manual MRZ info screen.
+                Toast.makeText(this, "Not implemented yet :(", Toast.LENGTH_SHORT).show()
```

Comment:
Optional: It seems we have several toasts here. How about bringing it to the extension? 😄

Reply:
@lydiasama they'll be removed once we implement the navigation, and since this is a sample, I don't think it's a good idea to make the devs spend time exploring the codebase (looking at `extensions` for example) to understand the usage of the library. Someone with the most rudimentary understanding of the Android framework should be able to understand the sample. What do you think?